### ISSUE TITLE
How to decode multiple objects in a single buffer?

### ISSUE THREAD

**Issue by Bilge** [kriszyp/msgpackr#14]
For example, currently the library behaves this way.

```
        const buf = new Uint8Array([1, 2]);
        console.log(unpack(buf)); // 1
        console.log(unpack(buf)); // 1
```

But we would like it to behave the following way.

```
        const buf = new Uint8Array([1, 2]);
        console.log(unpack(buf)); // 1
        console.log(unpack(buf)); // 2

**Comment by kriszyp** [comment]
Typically you would do this with streaming:
```
const parseStream = new UnpackrStream()
parseStream.on('data', (value) => {
	// called for each value, once for 1, then 2, in this example
})
parseStream.on('end', () => {/* end of stream */})
// stream in data, this is using the 
let bufferStream = new require('stream').Duplex() // create a simple stream
bufferStream.pipe(parseStream) // pipe to the parser
bufferStream.push(new Uint8Array([1, 2])); // stream in some data
bufferStream.push(null); // done
```
If you are looking at specifically parsing a single well-defined buffer/block that holds multiple values (with no values that span different chunks) with a more targeted API, open to considering something like that.

**Comment by Bilge** [comment]
I don't think I can use streams in the browser. This looks like Node.js type stuff.

**Comment by kriszyp** [comment]
No, you can't, you are right, this is using Node.js streams. It certainly is feasible to provide something more suitable to browsers. In your case I assume there is well-defined framing (from an HTTP response or WebSocket message), so no need to worry about values that span different chunks/buffers?

**Comment by Bilge** [comment]
In my case, the message is passed in the URL, so no; no need to worry about chunks. However, I just managed to get `@msgpack/msgpack` working instead, although the code is rather ugly and I don't know how portable it is but it works in modern browsers.

```js
        const decoded: number[] = [...];

        for await (const item of decodeStream(new ReadableStream({
            pull: controller => {
                controller.enqueue(decoded);
                controller.close();
            }
        }))) {
            console.debug(item);
        }
```

**Comment by kriszyp** [comment]
Ok, I added an `unpackMultiple` function, does this make sense for what you are trying to do?:
https://github.com/kriszyp/msgpackr#reading-multiple-values

**Comment by Bilge** [comment]
Yes, I believe so. Thanks!

**Comment by kriszyp** [comment]
Ok, published in v1.2.1
